source/agora/cli/client/SendTxProcess.d

```diff
@@ -225,13 +226,13 @@ unittest
         /// Contains the transaction cache
         private Transaction[Hash] tx_cache;
 
-        public override void putTransaction (Transaction tx) @safe
+        public override void putTransaction (in Transaction tx) @safe
         {
-            this.tx_cache[hashFull(tx)] = tx;
+            this.tx_cache[hashFull(tx)] = tx.serializeFull().deserializeFull!Transaction;
```

Comment:
Why do we serialize and deserialize the transaction?

Reply:
Good question..

Reply:
To deep copy it. That's the only way to get a mutable instance from a const one. And that's the only place where we need one.